File: src/test/java/net/bramp/ffmpeg/progress/RecordingProgressListener.java
package net.bramp.ffmpeg.progress;

import com.google.common.collect.Lists;
import java.util.List;

/** Test class to keep a record of all progresses. */
public class RecordingProgressListener implements ProgressListener {
  public final List<Progress> progresses = Lists.newArrayList();

  @Override
  public void progress(Progress p) {
    progresses.add(p);
  }

  public void reset() {
    progresses.clear();
  }
}

File: src/test/kotlin/net/bramp/ffmpeg/progress/RecordingProgressListener.kt
package net.bramp.ffmpeg.progress

/** Test class to keep a record of all progresses. */
class RecordingProgressListener : ProgressListener {
  val progresses = mutableListOf<Progress>()

  override fun progress(p: Progress) {
    progresses.add(p)
  }

  fun reset() {
    progresses.clear()
  }
}
